### ISSUE TITLE
Not working without passing CMSampleBuffer

### ISSUE THREAD

**Issue by chaudharyvivek** [pixiv/webrtc#26]
This works fine when I am passing CMSampleBuffer using deliverRecordedData. But in some case, I want it to handle automatically like GoogleWebRTC does. So this is not working for me without passing the sample buffer.

**Comment by akihikodaki** [comment]
It is a limitation of iOS.

**Comment by chaudharyvivek** [comment]
Is there not any option of setting useManualAudio = NO ? When I set it, I receive nothing on the web portal. So ideally, this is not an extension of GoogleWebRTC, It is a modified version of that with ONLY manual audio support. Am I right?

**Comment by akihikodaki** [comment]
It wouldn't work if you set YES to `useManualAudio` and NO to `isAudioEnabled` in broadcast extension. It should work in a normal application.

**Comment by chaudharyvivek** [comment]
I am not sure if I am able to explain my issue. Here is more explanation. I have two cases to achieve-
Case 1 - I need to pass my own audio buffers via webRTC. I have successfully achieved this via this extension.
Case 2 - I need to use it exactly same as GoogleWebRTC i.e. I don't want to pass audio buffers manually. I am unable to achieve this via this extension. Is this the limitation?

Thanks for your help.

**Comment by akihikodaki** [comment]
Yes, that is the limitation of broadcast extension. `useManualAudio` and `isAudioEnabled` are exposed just to make it compatible with both of the extension and normal application.

**Comment by chaudharyvivek** [comment]
I have a normal working application (iOS) where GoogleWebRTC is integrated and working fine. I just replaced GoogleWebRTC framework with the framework at this link - https://github.com/pixiv/webrtc/releases
My normal application stopped working. It only works if I provide sample buffer.
Will my normal application stop working with this framework?

**Comment by akihikodaki** [comment]
No, it should work. I have no idea why your application stopped working.

**Comment by chaudharyvivek** [comment]
Our project implementation is same as this - https://github.com/stasel/WebRTC-iOS
It works fine with GoogleWebRTC but not with this pixiv extension. Our project implementation works fine with the pixiv extension only if we pass buffer manually. Is there any additional setting or we need to set any additional property to make it work in automatic mode (without passing the manual CMSampleBuffer)?

**Comment by akihikodaki** [comment]
No. You have to debug your application with pixiv extension to see what went wrong.

**Comment by chaudharyvivek** [comment]
Can you please provide the steps to build the framework from source?

**Comment by chaudharyvivek** [comment]
Can you please provide the steps to compile the ObjC framework from source?

**Comment by shenfusheng** [comment]
@chaudharyvivek Could you share your code about how to send CMSampleBuffer using deliverRecordedData? I am blocked here, when I use deliverRecordedData, there will be an error that I mentioned here.[https://github.com/pixiv/webrtc/issues/30](url)

**Comment by nmalevich** [comment]
Hi!
Any solution for this problem?
I have the same issue

How should I configure audioSession in common iOS app? I have implemented that on broadcast extension
Thanks for your answers

**Comment by shenfusheng** [comment]
你好，邮件已收到，谢谢！
